pkg/spi/application/types.go

```diff
@@ -0,0 +1,33 @@
+package application
+
+import (
+	"github.com/docker/infrakit/pkg/types"
+)
+
+// Type is the type of an application.  This gives hint about what struct types to map to, etc.
+// It also marks one instance of an application as of different nature from another.
+type Type string
+
+//Operation : update operation code
+type Operation int
+
+const (
+	// TypeError is the type to use for sending errors in the transport of the events.
+	TypeError = Type("error")
+
+	// ADD new resources
+	ADD Operation = iota
+	// DELETE resources
+	DELETE
+	// UPDATE resources
+	UPDATE
+	// GET resources
+	GET
+)
```

Comment:
>  are you intending for this to be a way to do generic CRUD operations on objects? 

Yes. Since application has not been defined for a specific purpose, we think that it needs a general interface.

> What are objects that we are doing CRUD operations on?

Although it is sufficient to think of handling Group specs, it is not clear what kinds of objects to handle at the moment. Because the purpose of application is not specific.
If it is limited to Group spec, instance Spec, implementation of event repeater becomes complicated 
For that reason, we are currently setting it as a simple json object, of course GroupSpec could also be passed as a json object.

>I think this should be a RESTful API listening on a port 
> So you could deploy a web application …

As you say, I think the Restful interface is useful.
In my understanding, Is the infrakit application to connect deployed app (not infrakit app, app on docker cluster)  and Infrakit flavor and instance plugin for autoscale and rolling update?
If so it is quite consistent with my opinion.
I'd like to implement it as the next Infarakit application.